sycl/source/detail/device_image_impl.hpp

```diff
@@ -311,8 +311,15 @@ class device_image_impl {
   ~device_image_impl() {
     try {
       if (MProgram) {
+#ifdef _WIN32
+        if (!sycl::detail::GlobalHandler::instance().isUrTearDowned) {
+          const PluginPtr &Plugin = getSyclObjImpl(MContext)->getPlugin();
+          Plugin->call(urProgramRelease, MProgram);
```

Comment:
would it be possible for the call function to internally check `isUrTearDowned` and return some 'unitinitalized' value (or just ignore it)? That way, we generically workaround this issue in one place, rather than doing in on a case-by-case basis.